composer/callbacks/checkpoint_saver.py

```diff
@@ -303,11 +304,24 @@ def epoch_checkpoint(self, state: State, logger: Logger):
                 logger,
             )
 
-    def get_state_dict(self, state):
-        return {
-            'state': state.state_dict(),
-            'rng': reproducibility.get_rng_state(),
-        }
+    def state_dict(self) -> Dict[str, Any]:
+        state_dict = super().state_dict()
+
+        all_checkpoints = []
+        for save_filename, timestamp in self.all_saved_checkpoints_to_timestamp.items():
+            all_checkpoints.append((save_filename, timestamp.state_dict()))
+
+        # TODO: consider saving additional state for checkpoint rotation
+        state_dict['all_saved_checkpoints_to_timestamp'] = all_checkpoints
+        return state_dict
+
+    def load_state_dict(self, state: Dict[str, Any]):
+        super().load_state_dict(state)
+        if 'all_saved_checkpoints_to_timestamp' in state:
+            for (save_filename, timestamp_state) in state['all_saved_checkpoints_to_timestamp']:
+                new_timestamp = Timestamp()
+                new_timestamp.load_state_dict(timestamp_state)
+                self.all_saved_checkpoints_to_timestamp[save_filename] = new_timestamp
```

Comment:
```suggestion
                load_timestamp = Timestamp()
                load_timestamp.load_state_dict(timestamp_state)
                self.all_saved_checkpoints_to_timestamp[save_filename] = load_timestamp
```
so it's more clear it's not a "new" timestamp but rather loading the old one?

Reply:
I wanted something like `Timestamp.load_from_state(state) -> Timestamp` but realizing this could technically be `Timestamp(**timestamp_state)`? `load_state_dict` still feels a little safer for backwards/forwards compatible support

Reply:
yea I think load_state_dict is safer